src/utils.js

```diff
@@ -66,3 +66,65 @@ export const objectQueryString = obj => {
     .map(([k, v]) => `${k}:${serializeAttribute(v)}`)
     .join(';');
 };
+
+const strSplitOnFirst = (s, char) => {
+  const i = s.indexOf(char);
+  if (i === -1) {
+    return [s];
+  }
+
+  return [s.substring(0, i), s.substring(i + 1)];
+};
+
+/**
+   Compute longest common path prefix for an array of path strings.
+   The input array must be sorted lexically.
+*/
+const longestPathPrefix = sortedPaths => {
+  let first = sortedPaths[0];
+  let last = sortedPaths[sortedPaths.length - 1];
+  let result = '';
+
+  if (sortedPaths.length === 0) {
+    return result;
+  }
+  if (sortedPaths.length === 1) {
+    const comps = sortedPaths[0].split('/');
+    if (comps.length === 1) {
+      return result;
+    }
+    const path = comps.slice(0, comps.length - 1).join('/');
+    return `${path}/`;
+  }
+
+  while (first && last) {
+    const [firstCurrent, firstRest] = strSplitOnFirst(first, '/');
```

Comment:
Comment: I was thinking that it could make this function a bit simpler (and no need for `strSplitOnFirst`) if you'd `split` the paths on caller side. So that this function would receive not paths, but splitter path parts as parameter.

Reply:
True. That's technically more operations on strings upfront, but I guess speed is not of the essence here, as we have a small number of paths.